lib/apiservers/engine/backends/container.go

```diff
@@ -506,3 +663,185 @@ func (c *Container) imageExist(imageID string) (storeName string, err error) {
+	// FIXME: This is a temporary workaround until we have a name resolution story.
+	// Call imagec with -resolv parameter to learn the name of the vmdk and put it into in-memory map
+	cmdArgs := []string{"-reference", image, "-resolv", "-standalone", "-destination", os.TempDir()}
+
+	out, err := exec.Command(Imagec, cmdArgs...).Output()
+	if err != nil {
+		log.Printf("%s exit code: %s", Imagec, err)
+		return nil,
+			derr.NewErrorWithStatusCode(fmt.Errorf("Container look up failed"),
+				http.StatusInternalServerError)
+	}
+	var v1 struct {
+		ID string `json:"id"`
+		// https://github.com/docker/engine-api/blob/master/types/container/config.go
+		Config container.Config `json:"config"`
+	}
+	if err := json.Unmarshal(out, &v1); err != nil {
+		return nil,
+			derr.NewErrorWithStatusCode(fmt.Errorf("Failed to unmarshall image history: %s", err),
+				http.StatusInternalServerError)
+	}
+	log.Printf("v1 = %+v", v1)
+
+	imageMetadata := &viccontainer.VicContainer{
+		ID:     v1.ID,
+		Config: &v1.Config,
+	}
+
+	return imageMetadata, nil
+}
+
+// attacheStreams takes the the hijacked connections from the calling client and attaches
+// them to the 3 streams from the portlayer's rest server.
+// name is the container id
+// tty indicates whether we should use Docker's copyescapable for stdin
+// stdinOnce indicates whether to close the stdin at the container when we finish (we don't handle this but still need to check)
+// clStdin, clStdout, clStderr are the hijacked connection
+// keys are the keys that are used for detaching streams without closing them
+func attachStreams(name string, tty, stdinOnce bool, clStdin io.ReadCloser, clStdout, clStderr io.Writer, keys []byte) error {
+	var wg sync.WaitGroup
+
+	errors := make(chan error, 3)
+	//FIXME: Swagger will timeout on us.  We need to either have an infinite timeout or the timeout should
+	// start after some inactivity?
+	inContext, inCancel := context.WithTimeout(context.Background(), 2*time.Hour)
+	outContext, outCancel := context.WithTimeout(context.Background(), 2*time.Hour)
+	errContext, errCancel := context.WithTimeout(context.Background(), 2*time.Hour)
+
+	if clStdin != nil {
+		wg.Add(1)
+		// Pipe for stdin so we can interject and watch the input streams for detach keys.
+		stdinReader, stdinWriter := io.Pipe()
+
+		defer clStdin.Close()
+		defer stdinReader.Close()
+
+		transport := httptransport.New(PortLayerServer(), "/", []string{"http"})
+		plClient := client.New(transport, nil)
+		transport.Consumers["application/octet-stream"] = httpkit.ByteStreamConsumer()
+		transport.Producers["application/octet-stream"] = httpkit.ByteStreamProducer()
+
+		// Swagger wants an io.reader so give it the reader pipe.  Also, the swagger call
+		// to set the stdin is synchronous so we need to run in a goroutine
+		go func() {
+			setStdinParams := interaction.NewContainerSetStdinParamsWithContext(inContext).WithID(name)
```

Comment:
really?  Isn't that a little pedantic?  Besides, I like to keep code usage together.  Unless you have a really good reason,...